tools/local_fix_format.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh
+#
+# Runs the Enovy format fixers on the current changes. By default, this runs on
+# changes that are not yet committed. You can also specify:
+#
+#   -all: runs on the entire repository (very slow)
+#   -main: runs on the changes amde from the main branch
+#   file1 file2 file3...: runs on the specified files.
+#
+# To effectively use this script, it's best to run the format fixer before each
+# call to 'git commit'.  If you forget to do that then you can run with "-main"
+# and it will run on all changes you've made since branching from main, which
+# will still be relatively fast.
+#
+# To run this script, you must provide several environment variables:
+#
+# CLANG_FORMAT     -- points to the clang formatter binary, perhaps found in
+#                     $CLANG_TOOLCHAIN/bin/clang-format
+# BUILDIFIER_BIN   -- buildifier location, perhaps found in $HOME/go/bin/buildifier
+# BUILDOZER_BIN    -- buildozer location, perhaps found in $HOME/go/bin/buildozer
+
+# If DISPLAY is set, then tkdiff pops up for some BUILD changes.
+unset DISPLAY
+
+# Trigger an error if the required env vars are not set.
+set -u
+ignore="$CLANG_FORMAT $BUILDIFIER_BIN $BUILDOZER_BIN"
+
+if [[ $# > 0 && "$1" == "-all" ]]; then
+  echo "Checking all files in the repo...this may take a while."
+  shift
+  args="$@"
+else
+  if [[ $# > 0 && "$1" == "-main" ]]; then
+    shift
+    echo "Checking all files that have changed since the main branch."
+    args=$(git diff main | grep ^diff | awk '{print $3}' | cut -c 3-)
+  elif [[ $# == 0 ]]; then
+    args=$(git status|egrep '(modified:|added:)'|awk '{print $2}')
+    args+=$(git status|egrep 'new file:'|awk '{print $3}')
+  else
+    args="$@"
+  fi
```

Comment:
Use `./tools/git/modified_since_last_github_commit.sh`? proto_format does same so it provides consistency between toolings.

Reply:
I was not able to get that to produce the same results. Its implementation is certainly completely different, but for me it generates nothing when I run it on the client where I'm editing this.

Reply:
it greps for file extensions - perhaps that is why

Reply:
I modified a .cc file and it did not catch it. Is there some trick to calling it?

Reply:
not sure - i havent really used it directly - just hacked the code that does

```console
$ git grep since_last_github
tools/code_format/format_python_tools.sh:"$(dirname "$0")"/../git/modified_since_last_github_commit.sh ./ py || \
tools/proto_format/proto_format.sh:./tools/git/modified_since_last_github_commit.sh ./api/envoy proto || \
```

Reply:
well I think I'll leave this unresolved as what I have works for me, and (for whatever reason) the other script does not.